src/header/common/if_match.rs

```diff
@@ -0,0 +1,82 @@
+use header::{EntityTag, Header, HeaderFormat};
+use header::parsing;
+use std::fmt;
+use std::str::FromStr;
+
+/// An entity tag match specification
+#[derive(PartialEq, Clone, Debug)]
+pub enum EntityTagMatch{
+    EntityTags(Vec<EntityTag>),
+    Star
+}
+
+/// The `If-Match` header
+///
+/// The `If-Match` header can be used by clients to indicate what
+/// response matchs they if.
+#[derive(Clone, PartialEq, Debug)]
+pub struct IfMatch(pub EntityTagMatch);
+
+deref!(IfMatch => EntityTagMatch);
+
+impl Header for IfMatch {
+    fn header_name() -> &'static str {
+        "If-Match"
+    }
+
+    fn parse_header(raw: &[Vec<u8>]) -> Option<IfMatch> {
+        if raw.get(0).and_then(|x| x.get(0).map(|y| y == &b'*')).unwrap_or(false) {
+            Some(IfMatch(EntityTagMatch::Star))
+        } else {
+            match parsing::from_comma_delimited(raw).map(EntityTagMatch::EntityTags) {
+                Some(tag) => Some(IfMatch(tag)),
+                None => None
+            }
+        }
+    }
+}
+
+impl HeaderFormat for IfMatch {
+    fn fmt_header(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self.0 {
+            EntityTagMatch::Star => write!(fmt, "{}", "*"),
+            EntityTagMatch::EntityTags(ref x) =>
+                parsing::fmt_comma_delimited(fmt, &x[])
+        }
+    }
+}
+
+impl FromStr for IfMatch {
+    type Err = ();
+    fn from_str(s: &str) -> Result<IfMatch, ()> {
+        if s.trim()=="*" {
+            Ok(IfMatch(EntityTagMatch::Star))
+        } else {
+            let mut v : Vec<u8> = Vec::new();
+            v.push_all(s.as_bytes());
+            match parsing::from_comma_delimited(&[v]).map(EntityTagMatch::EntityTags) {
+                Some(tag) => Ok(IfMatch(tag)),
+                None => Err(())
+            }
+        }
+    }
+}
+
+#[test]
```

Comment:
Can you add some benches too like all the other headers? They are generated using a macro.